smart-contracts/contracts/mixins/MixinTransfer.sol

```diff
@@ -44,6 +44,65 @@ contract MixinTransfer is
   // This is calculated as `keyPrice * transferFeeBasisPoints / BASIS_POINTS_DEN`.
   uint public transferFeeBasisPoints;
 
+  /**
+  * @notice Allows the key owner to share their key (parent key) by
+  * transferring a portion of the remaining time to a new key (child key).
+  * @param _from The owner of the parent key
+  * @param _to The recipient of the shared key
+  * @param _tokenId the key to share
+  * @param _timeShared The amount of time shared
+  */
+  function shareKey(
+    address _from,
+    address _to,
+    uint _tokenId,
+    uint _timeShared
+  ) public
+    onlyIfAlive
+    hasValidKey(_from)
+    onlyKeyOwnerOrApproved(_tokenId)
+  {
+    require(_to != address(0), 'INVALID_ADDRESS');
+
+    Key storage fromKey = keyByOwner[_from];
+    Key storage toKey = keyByOwner[_to];
+    uint IdTo = toKey.tokenId;
+    uint time;
+    // get the remaining time for the origin key
+    uint timeRemaining = fromKey.expirationTimestamp - block.timestamp;
+    // get the transfer fee
+    uint fee = getTransferFee(_from, _timeShared);
+    uint totalTime = _timeShared + fee;
+
+    // ensure that we don't try to share too much
+    if(totalTime < timeRemaining) {
+      time = totalTime;
+      // deduct time from parent key, including transfer fee
+      _timeMachine(_tokenId, time, false);
+    } else {
+      time = timeRemaining - fee;
+      fromKey.expirationTimestamp = block.timestamp; // Effectively expiring the key
+      emit ExpireKey(_tokenId);
+    }
+
+    if (toKey.tokenId == 0) {
+      _assignNewTokenId(toKey);
+      _recordOwner(_to, IdTo);
+    }
+    // add time to new key
+    _timeMachine(IdTo, time, true);
```

Comment:
Hmm this is transferring `_timeShared + fee` to the other account?  For this maybe we should add a test that confirms the total time remaining across both accounts is <= the original time remaining - fee

Reply:
Good catch. I've reworked this to handle this better, but I'll add a test case as you suggest as well.